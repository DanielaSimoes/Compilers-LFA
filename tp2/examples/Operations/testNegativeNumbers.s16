program finalTest ()
{
    int y = 3;
    int n = -22;
    float x = -2.35;
    // assignment with vars
    n = y;

    exit();
}
