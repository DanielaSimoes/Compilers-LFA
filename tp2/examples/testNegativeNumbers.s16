program finalTest ()
{
    int y = 3;
    int n = -22;
    // assignment with vars
    n = y;

    exit();
}
