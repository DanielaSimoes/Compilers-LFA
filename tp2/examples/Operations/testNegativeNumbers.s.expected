               ; declarating a integer variable
               .data
y:             .word 3

               ; declarating a integer variable
n:             .word -22

               ; declarating a float variable
x:             .float -2.350000

               ; assigning value to n
               .text
L0:
               ; checking value of variable
               load y

               store n

               ; exiting program
               halt

               ; includes
               halt
printInt:
               ipush 28
printInt1:     dup2
               iushr
               ipush 0xF
               iand
               jsr pNibble
               dup
               ifeq printInt2
               ipush 4
               isub
               goto printInt1
printInt2:     pop
               ret

pNibble:
               ipush 0x30
               iadd
               dup
               ipush 0x39
               isub
               ifle pNibble1
               ipush 0x0a
               iadd
pNibble1:      write
               ret

printStr:      dup
               ipush 0
               baload
               dup
               ifeq printStr_end
               write
               ipush 1
               iadd
               goto printStr

printStr_end:  write
               ret
